tests/core/string/test_string.h

```diff
@@ -456,30 +456,79 @@ TEST_CASE("[String] Number to string") {
-	static const int64_t num[4] = { 0xFFAE, 0x22, 0, 0xAADDAD };
+	static const char *nums[12] = { "0xFFAE", "22", "0", "AADDAD", "0x7FFFFFFFFFFFFFFF", "-0xf", "", "000", "000f", "0xaA", "-ff", "-" };
+	static const int64_t num[12] = { 0xFFAE, 0x22, 0, 0xAADDAD, 0x7FFFFFFFFFFFFFFF, -0xf, 0, 0, 0xf, 0xaa, -0xff, 0x0 };
 
-	for (int i = 0; i < 4; i++) {
+	for (int i = 0; i < 12; i++) {
 		CHECK(String(nums[i]).hex_to_int() == num[i]);
 	}
+
+	// Invalid hex strings should return 0.
+	static const char *invalid_nums[15] = { "qwerty", "QWERTY", "0xqwerty", "0x00qwerty", "qwerty00", "0x", "0x__", "__", "x12", "+", " ff", "ff ", "f f", "+ff", "--0x78" };
+
+	ERR_PRINT_OFF
+	for (int i = 0; i < 15; i++) {
+		CHECK(String(invalid_nums[i]).hex_to_int() == 0);
+	}
+
+	CHECK(String("0xFFFFFFFFFFFFFFFFFFFFFFF").hex_to_int() == INT64_MAX); // Too large, largest possible is returned.
+	CHECK(String("-0xFFFFFFFFFFFFFFFFFFFFFFF").hex_to_int() == INT64_MIN); // Too small, smallest possible is returned.
+	ERR_PRINT_ON
+}
+
+TEST_CASE("[String] Bin to integer") {
+	static const char *nums[10] = { "", "0", "0b0", "0b1", "0b", "1", "0b1010", "-0b11", "-1010", "0b0111111111111111111111111111111111111111111111111111111111111111" };
+	static const int64_t num[10] = { 0, 0, 0, 1, 0, 1, 10, -3, -10, 0x7FFFFFFFFFFFFFFF };
+
+	for (int i = 0; i < 10; i++) {
+		CHECK(String(nums[i]).bin_to_int() == num[i]);
+	}
+
+	// Invalid bin strings should return 0. The long "0x11...11" is just too long for a 64 bit int.
+	static const char *invalid_nums[16] = { "qwerty", "QWERTY", "0bqwerty", "0b00qwerty", "qwerty00", "0x__", "0b__", "__", "b12", "+", "-", "0x12ab", " 11", "11 ", "1 1", "--0b11" };
+
+	ERR_PRINT_OFF
+	for (int i = 0; i < 16; i++) {
+		CHECK(String(invalid_nums[i]).bin_to_int() == 0);
+	}
+
+	CHECK(String("0b111111111111111111111111111111111111111111111111111111111111111111111111111111111").bin_to_int() == INT64_MAX); // Too large, largest possible is returned.
+	CHECK(String("-0b111111111111111111111111111111111111111111111111111111111111111111111111111111111").bin_to_int() == INT64_MIN); // Too small, smallest possible is returned.
+	ERR_PRINT_ON
 }
 
 TEST_CASE("[String] String to float") {
-	static const char *nums[4] = { "-12348298412.2", "0.05", "2.0002", " -0.0001" };
-	static const double num[4] = { -12348298412.2, 0.05, 2.0002, -0.0001 };
+	static const char *nums[12] = { "-12348298412.2", "0.05", "2.0002", " -0.0001", "0", "000", "123", "0.0", "000.000", "000.007", "234__", "3..14" };
+	static const double num[12] = { -12348298412.2, 0.05, 2.0002, -0.0001, 0.0, 0.0, 123.0, 0.0, 0.0, 0.007, 234.0, 3.0 };
 
-	for (int i = 0; i < 4; i++) {
+	for (int i = 0; i < 12; i++) {
 		CHECK(!(ABS(String(nums[i]).to_float() - num[i]) > 0.00001));
 	}
+
+	// Invalid float strings should return 0.
+	static const char *invalid_nums[6] = { "qwerty", "qwerty123", "0xffff", "0b1010", "--3.13", "__345" };
+
+	ERR_PRINT_OFF
```

Comment:
(my bad was meant here)
Is this really required? From what I can see there's no errors on this thrown

Reply:
You are correct, only `hex_to_int()` prints errors when the string contains illegal characters and 0 is returned. Fixed this test and "Bin to integer" test. Also added the "too large exponent" tests.

Reply:
Now the unnecessary `ERR_PRINT_OFF`'s are removed and "too high exponent" tests are added.

Reply:
Note that this will now print a warning cluttering the test output, see once it's run

Reply:
The too high exponent case isn't triggered here as it fires with exponent of 512, unsure why it's still infinity here though

Reply:
Fixed this. There seems to be different levels of "too high exponents". Floats between 1e309 - 1e511 will become `INFINITY`, but warning is not printed (`hex_to_int` prints errors, btw). When the float gets even larger, 1e512, warning is finally printed. Don't ask me why floats work like this :)